src/evaluator.ts

```diff
@@ -982,10 +996,12 @@ class Evaluator {
 
         metrics.cost += row.cost || 0;
 
-        await runExtensionHook(testSuite.extensions, 'afterEach', {
-          test: evalStep.test,
-          result: row,
-        });
+        if (hasDefinedExtensionHooks(testSuite.extensions)) {
+          await runExtensionHook(testSuite.extensions!, 'afterEach', {
+            test: evalStep.test,
+            result: row,
+          });
+        }
```

Comment:
wondering if we should move the check for `hasDefinedExtensionHooks` into `runExtensionHook`- would make the flow in this function cleaner and it seems like there must be a check for if `afterEach`/other hook is present in the extensions or not?

Reply:
That's actually how it was originally (returning an empty array). IMO, we should assume that whatever is returned from `runExtensionHook` – aside from `undefined` – was returned by the underlying hook call. I could move this check into `runExtensionHook` and return `undefined` but might then want to coerce that to avoid the `undefined` issue I mentioned above.